### ISSUE TITLE
Parser fails on lifetimes in the parameter list and return type

### ISSUE THREAD

**Issue by parno** [jaybosamiya/verusfmt#15]
This example:
```
fn get(&'a self) -> (o: &'a V) {
    0
}
```
fails with:
```
1 | fn get(&'a self) -> (o: &'a V) {
  |                           ^---
  |
  = expected COMMENT, colon_str, dot_dot_eq_str, dot_dot_str, or pipe_str
```

but the following work:
```
fn get(&'a self) -> (o: V) {
    0
}
```
```
fn get(a:int) -> (o: &'a V) {
    0
}
```
```
fn get(&'a self, b:int) -> (o: &'a V) {
    0
}
```
And on its own, `-> (o: &'a V)` is correctly parsed as a `ret_type`.

This would be easier to debug if we could get a partial parse tree out of Pest when it fails.

**Comment by jaybosamiya** [comment]
Unfortunately it doesn't seem like there is an easy way to get a partial parse tree out of Pest.

However, if we just keep repeatedly adding one of the "expected" at the point it expects it, then we can force it to give a complete tree, and this helps identify the issue and fix it.

For example, taking the above example and completing it to something the parser (before the fix) would finish parsing, we get:
```
fn get(&'a self) -> (o: &':a !V) {
    0
}
```
which parses to
```
- fn
  - fn_str: "fn"
  - name > identifier: "get"
  - param_list > param
    - pat_no_top_alt > pat_inner > ref_pat > pat > pat_inner > literal_pat > literal > char: "'a self) -> (o: &'"
    - colon_str: ":"
    - type > macro_type > macro_call
      - path > path_segment > name_ref > identifier: "a"
      - bang_str: "!"
      - token_tree > token > identifier: "V"
  - fn_qualifier: ""
  - fn_terminator > fn_block_expr > expr > expr_inner > literal > int_number: "0"
```

Clearly that `char` there is just bogus, so I looked at the `char` definition and fixed it in 4490ebeeb52e1695e9566625bd9c3a16e22053ce

**Comment by parno** [comment]
Ah, nice find.  Thanks for tracking it down and for the explanation.
